python/tskit/tables.py

```diff
@@ -43,54 +42,55 @@
 import tskit.util as util
 from tskit import UNKNOWN_TIME
 
+
 dataclass_options = {"frozen": True}
 
 
-@dataclass(eq=False, **dataclass_options)
+@metadata.lazy_decode
+@dataclass(**dataclass_options)
 class IndividualTableRow:
     __slots__ = ["flags", "location", "parents", "metadata"]
     flags: int
     location: np.ndarray
     parents: np.ndarray
-    metadata: bytes
+    metadata: Any
 
     def __eq__(self, other):
-        if not isinstance(other, type(self)):
-            return False
-        else:
-            return all(
-                (
-                    self.flags == other.flags,
-                    np.array_equal(self.location, other.location),
-                    np.array_equal(self.parents, other.parents),
-                    self.metadata == other.metadata,
-                )
-            )
+        return (
+            isinstance(other, IndividualTableRow)
+            and self.flags == other.flags
+            and np.array_equal(self.location, other.location)
+            and np.array_equal(self.parents, other.parents)
+            and self.metadata == other.metadata
+        )
 
     def __neq__(self, other):
```

Comment:
Two things here:

1. I think we can get rid of ``__neq__`` now, this is handled automatically in more recent versions of Python
2. Can we use the dataclass ``__eq__``? If not, let's add a comment saying why, as this will be a recurring question.

(Ah, answering my own question here: we have numpy arrays. Good to add a quick comment here to that effect)